contracts/src/v0.8/vrf/dev/VRFV2PlusWrapper.sol

```diff
@@ -191,46 +201,59 @@ contract VRFV2PlusWrapper is ConfirmedOwner, TypeAndVersionInterface, VRFConsume
    * @param _fallbackWeiPerUnitLink is the backup LINK exchange rate used when the LINK/NATIVE feed
    *        is stale.
    *
-   * @param _fulfillmentFlatFeeLinkPPM is the flat fee in millionths of LINK that VRFCoordinatorV2Plus
-   *        charges.
-   *
    * @param _fulfillmentFlatFeeNativePPM is the flat fee in millionths of native that VRFCoordinatorV2Plus
-   *        charges.
+   *        charges for native payment.
+   *
+   * @param _fulfillmentFlatFeeLinkDiscountPPM is the flat fee discount in millionths of native that VRFCoordinatorV2Plus
+   *        charges for link payment.
    */
   function setConfig(
     uint32 _wrapperGasOverhead,
     uint32 _coordinatorGasOverhead,
-    uint8 _wrapperPremiumPercentage,
+    uint8 _wrapperNativePremiumPercentage,
+    uint8 _wrapperLinkPremiumPercentage,
     bytes32 _keyHash,
     uint8 _maxNumWords,
     uint32 _stalenessSeconds,
     int256 _fallbackWeiPerUnitLink,
-    uint32 _fulfillmentFlatFeeLinkPPM,
-    uint32 _fulfillmentFlatFeeNativePPM
+    uint32 _fulfillmentFlatFeeNativePPM,
+    uint32 _fulfillmentFlatFeeLinkDiscountPPM
   ) external onlyOwner {
+    if (_fulfillmentFlatFeeLinkDiscountPPM > _fulfillmentFlatFeeNativePPM) {
+      revert LinkDiscountTooHigh(_fulfillmentFlatFeeLinkDiscountPPM, _fulfillmentFlatFeeNativePPM);
+    }
+    if (_wrapperNativePremiumPercentage > PREMIUM_PERCENTAGE_MAX) {
+      revert InvalidPremiumPercentage(_wrapperNativePremiumPercentage, PREMIUM_PERCENTAGE_MAX);
+    }
+    if (_wrapperLinkPremiumPercentage > PREMIUM_PERCENTAGE_MAX) {
+      revert InvalidPremiumPercentage(_wrapperLinkPremiumPercentage, PREMIUM_PERCENTAGE_MAX);
+    }
+
     s_wrapperGasOverhead = _wrapperGasOverhead;
     s_coordinatorGasOverhead = _coordinatorGasOverhead;
-    s_wrapperPremiumPercentage = _wrapperPremiumPercentage;
+    s_wrapperNativePremiumPercentage = _wrapperNativePremiumPercentage;
```

Comment:
Didn't we discuss removing wrapper premium completely and just having a premium in coordinator?

Reply:
This is kind of the coordinator's premium. This `wrapperXPremiumPercentage` has to mirror the value's set in the coordinator.

The wrapper's calculate price is roughly `wrapper_gas + (coordinator_gas + consumer_gas) * premium + flat`, simplified.